From b8f123577e1e42fd6f3dc5402df9c65236f532a8 Mon Sep 17 00:00:00 2001
From: Georgios Kourachanis <gkourachanis@users.noreply.github.com>
Date: Sat, 9 Jul 2022 19:32:32 +0300
Subject: [PATCH] Fix documentation widgets in GNU Octave v7.1.0

Documentation widgets got their functions renamed in GNU Octave v7.1.0 release.
This commit updates those functions' names for octave-tablicious.

Signed-off-by: Georgios Kourachanis <geo.kourachanis@gmail.com>
---
 inst/+tablicious/+internal/load_tablicious.m   | 4 +++-
 inst/+tablicious/+internal/unload_tablicious.m | 4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/inst/+tablicious/+internal/load_tablicious.m b/inst/+tablicious/+internal/load_tablicious.m
index 029d859..c3f55c0 100644
--- a/inst/+tablicious/+internal/load_tablicious.m
+++ b/inst/+tablicious/+internal/load_tablicious.m
@@ -37,8 +37,10 @@
   if ! isempty (qhelp_file)
     if compare_versions (version, "4.4.0", ">=") && compare_versions (version, "6.0.0", "<")
       __octave_link_register_doc__ (qhelp_file);
-    elseif compare_versions (version, "6.0.0", ">=")
+    elseif compare_versions (version, "6.0.0", ">=") && compare_versions (version, "7.1.0", "<")
       __event_manager_register_doc__ (qhelp_file);
+    elseif compare_versions (version, "7.1.0", ">=")
+      __event_manager_register_documentation__ (qhelp_file);
     endif
   endif
   
diff --git a/inst/+tablicious/+internal/unload_tablicious.m b/inst/+tablicious/+internal/unload_tablicious.m
index 7d2145a..bcaa44f 100644
--- a/inst/+tablicious/+internal/unload_tablicious.m
+++ b/inst/+tablicious/+internal/unload_tablicious.m
@@ -21,8 +21,10 @@
   if ! isempty (qhelp_file)
     if compare_versions (version, "4.4.0", ">=") && compare_versions (version, "6.0.0", "<")
       __octave_link_unregister_doc__ (qhelp_file);
-    elseif compare_versions (version, "6.0.0", ">=")
+    elseif compare_versions (version, "6.0.0", ">=") && compare_versions (version, "7.1.0", "<")
       __event_manager_unregister_doc__ (qhelp_file);
+    elseif compare_versions (version, "7.1.0", ">=")
+      __event_manager_unregister_documentation__ (qhelp_file);
     endif
   endif
   
